code/game/objects/items/stacks/cable.dm

```diff
@@ -168,6 +168,51 @@ var/global/list/datum/stack_recipe/cable_recipes = list ( \
 				getFromPool(/obj/item/stack/cable_coil, C.loc)
 				returnToPool(C)
 
+//I really wish that redundant shit wasn't needed, but snowflake code yeah
+/obj/item/stack/cable_coil/proc/catwalk_place(obj/structure/catwalk/Catwalk, mob/user)
```

Comment:
Except, like I said, you could have the catwalk attackby set up the place proc for the turf on which the catwalk is located, not the catwalk itself. This is not necessary. You even set the cable's loc to catwalk.loc.